src/Jss.js

```diff
@@ -45,9 +45,19 @@ export default class Jss {
    * Create a Style Sheet.
    */
   createStyleSheet(styles: Object, options: StyleSheetOptions): StyleSheet {
+    let generateClassName
+    if (options
+      && options.sheetsRegistryContext
+      && this.options.generateClassName === generateClassNameDefault) {
```

Comment:
The logic in that block, coupled with the `else` block and the `...options` spread below, is as follows:

Did a `sheetRegistryContext` get passed in, and is the Jss-instance top-level `generateClassName` still set to the original default?
* Yes? Then use the `generateClassName` from the registry context.
* No? Then use the `generateClassName` from the Jss instance and disregard the registry context.

Then, override all of that (with the options spread) if a `generateClassName` option got passed into `createStyleSheet`